tempesta_fw/classifier/frang.c

```diff
@@ -69,20 +69,54 @@ typedef struct {
 	unsigned int	req;
 } FrangRates;
 
+/* The least power of two greater than a year in seconds */
+#define FRANG_RESP_TIME_PERIOD 33554432
+
+/**
+ * Response code record.
+ *
+ * @cnt	- Amount of responses in a time frame part;
+ * @ts	- Response time in seconds. Four bytes could be used to store enough
+ *   number of seconds in the assumption that there won't be delays between user
+ *   responses longer than a year;
+ */
+typedef struct {
+	long	cnt;
+	int	ts;
+} __attribute__((packed)) FrangRespCodeStat;
```

Comment:
Without packing:
sizeof(FrangRespCodeStat) == 16; sizeof(resp_code_stat) == 128; sizeof(FrangAcc) == 264

With packing:
sizeof(FrangRespCodeStat) == 12; sizeof(resp_code_stat) == 96; sizeof(FrangAcc) == 232

It's needed to store @ts as int because @cnt could be greater than 2G and FrangAcc should be <= TFW_CLASSIFIER_ACCSZ == 256 bytes.

Reply:
`int ts` cause hacks, better to increase `TFW_CLASSIFIER_ACCSZ`. `TfwClassifierPrvt` is allocated once per client it's not a problem.

Reply:
I agree with @bartolootrit : there is no sense to loose the 4 bytes for each item. Frang is a DDoS mitigation module, so we have to spend system resources for malicious users. We can expect millions of users (keep in mind influence of IoT), so while RAM is chap nowadays we'll use CPU caches ineficiantly if we increase the data structure significantly. Moreover, we're going to introduce many other classification algorithms and will need to increase the accounting data structure anyway, so it's better to keep it as tiny as possible.